code/game/machinery/fat_sucker.dm

```diff
@@ -0,0 +1,212 @@
+/obj/machinery/fat_sucker
+	name = "lipid extractor"
+	desc = "Safely and efficiently extracts excess fat from a subject."
+	icon = 'icons/obj/machines/fat_sucker.dmi'
+	icon_state = "fat"
+
+	state_open = FALSE
+	density = TRUE
+	req_access = list(ACCESS_KITCHEN)
+	var/processing = FALSE
+	var/start_at = NUTRITION_LEVEL_WELL_FED
+	var/stop_at = NUTRITION_LEVEL_STARVING
+	var/free_exit = TRUE //set to false to prevent people from exiting before being completely stripped of fat
+	var/bite_size = 15 //amount of nutrients we take per process
+	var/nutrients //amount of nutrients we got build up
+	var/nutrient_to_meat = 90 //one slab of meat gives about 52 nutrition
+	var/datum/looping_sound/microwave/soundloop //100% stolen from microwaves
+	var/breakout_time = 600
+
+	var/next_fact = 10 //in ticks, so about 20 seconds
+	var/static/list/fat_facts = list(\
+	"Fats are triglycerides made up of a combination of different building blocks; glycerol and fatty acids.", \
+	"Adults should get a recommended 20-35% of their energy intake from fat.", \
+	"Being overweight or obese puts you at an increased risk of chronic diseases, such as cardiovascular diseases, metabolic syndrome, type 2 diabetes and some types of cancers.", \
+	"Not all fats are bad. A certain amount of fat is an essential part of a healthy balanced diet. " , \
+	"Saturated fat should form no more than 11% of your daily calories.", \
+	"Unsaturated fat, that is monounsaturated fats, polyunsaturated fats and omega-3 fatty acids, is found in plant foods and fish." \
+	)
+
+/obj/machinery/fat_sucker/Initialize()
+	. = ..()
+	soundloop = new(list(src),  FALSE)
+	update_icon()
+
+/obj/machinery/fat_sucker/RefreshParts()
+	..()
+	var/rating = 0
+	for(var/obj/item/stock_parts/micro_laser/L in component_parts)
+		rating += L.rating
+	bite_size = initial(bite_size) + rating * 5
+	nutrient_to_meat = initial(nutrient_to_meat) - rating * 5
+
+/obj/machinery/fat_sucker/examine(mob/user)
+	. = ..()
+	to_chat(user, "<span class='notice'>Alt-Click to toggle the safety hatch.</span>")
+	to_chat(user, "<span class='notice'>Removing [bite_size] nutritional units per operation.</span>")
+	to_chat(user, "<span class='notice'>Requires [nutrient_to_meat] nutritional units per meat slab.</span>")
+
+/obj/machinery/fat_sucker/close_machine(mob/user)
+	if(panel_open)
+		to_chat(user, "<span class='warning'>You need to close the maintenance hatch first!</span>")
+		return
+	..()
+	playsound(src, 'sound/machines/click.ogg', 50)
+	if(occupant)
+		if(!iscarbon(occupant))
+			occupant.forceMove(drop_location())
+			occupant = null
+			return
+		to_chat(occupant, "<span class='notice'>You enter [src]</span>")
+		addtimer(CALLBACK(src, .proc/start_extracting), 20, TIMER_OVERRIDE|TIMER_UNIQUE)
+		update_icon()
+
+/obj/machinery/fat_sucker/open_machine(mob/user)
+	make_meat()
```

Comment:
this proc is only called when you _open_ the machine?

Reply:
Open machine also runs when it automaticly opens

Reply:
oh I see it pulls nutrition from the mob on close but doesn't actually make the meat until it opens.